source/common/tls/server_context_impl.cc

```diff
@@ -95,12 +95,14 @@ ServerContextImpl::ServerContextImpl(Stats::Scope& scope,
                                      Ssl::ContextAdditionalInitFunc additional_init,
                                      absl::Status& creation_status)
     : ContextImpl(scope, config, factory_context, additional_init, creation_status),
-      tls_certificate_selector_(config.createTlsCertificateSelector()(config, *this)),
       session_ticket_keys_(config.sessionTicketKeys()),
       ocsp_staple_policy_(config.ocspStaplePolicy()) {
   if (!creation_status.ok()) {
     return;
   }
+  // After checked creation_status, so that the cert selector doesn't need to handle it.
```

Comment:
I don't understand what this comment is saying. Is the point simple that if creation failed, then don't create the selector?